### DISCUSSION TITLE
Verify subject Certificate?

### DISCUSSION THREAD

**Discussion by keskinadem** [strongswan/strongswan#2436] [Q&A] [answered]
In tunnel mode, initiator sends its own cert to responder for responder's verification in first IKE_AUTH(i) exchange but before that does initiator verify its own certificate before send it? If initiator verifies its own certificate before send IKE_AUTH message, is it up to any configuration?

**Comment by tobiasbrunner** [comment, marked as answer]
> but before that does initiator verify its own certificate before send it?

No.

**Reply by keskinadem** [reply]
My version is 5.6.0
My auth method is pubkey and initiator tries to verify its subject certificate with its CA cert before send IKE_AUTH 
- https://github.com/strongswan/strongswan/commit/fb63012e0cad81aa7a4289be6d6f1fea8bd2811f#diff-724528b4783e9d87c4d6c5c1647794478f5832f452cac1de215e028a84a6c565R430
- https://github.com/strongswan/strongswan/blob/master/src/libstrongswan/plugins/openssl/openssl_ec_public_key.c#L208

**Reply by tobiasbrunner** [reply]
It tries to build a trust chain (e.g. to collect intermediate CA certificates that might have to be sent to the peer), but it won't fail it that isn't successful.
